content/articles/2018-11-tibble-2.0.0-pre-announce.Rmd

```diff
@@ -0,0 +1,222 @@
+---
+
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>", error = TRUE)
+```
+
+Version 2.0.0 of the [tibble package](https://tibble.tidyverse.org) is almost ready for release. Tibbles are a modern reimagining of the data frame, keeping what time has shown to be effective, and throwing out what is not, with nicer default output too! Grab the development version with:
+
+```{r eval = FALSE}
+devtools::install_github("tidyverse/tibble")
+```
+
+We're making a pre-release announcement, because some changes require the attention of maintainers of packages that import or otherwise depend on tibble.
+This post describes how to adapt to the next version of tibble and is also an invitation for maintainers to provide feedback before v2.0.0 is finalized and submitted to CRAN.
+The easiest way to get in touch is to file an issue at https://github.com/tidyverse/tibble/issues (or to comment to an existing one).
+This blog post is aimed at package developers and those who maintain "production" scripts or apps. A high-level overview of new user-facing features will come in a separate blog post.
+
+## Reverse dependency checks
+
+We ran `R CMD check` for over 3000 CRAN and Bioconductor packages that depend directly or indirectly on the tibble package and compared results obtained with the CRAN versus development version of tibble.
+We will notify the maintainers of all affected packages (regardless of the check results of their package) and aim for a CRAN release before Christmas, so the dust has settled in time for [rstudio::conf](https://www.rstudio.com/conference/).
+
+We made pull requests to implement the necessary changes in several of the most heavily downloaded packages. Based on this experience, this post highlights the problems downstream maintainers are most likely to see and how to solve them. Most fixes should be quite simple.
+
+For the full list of changes, features, and bug fixes, please see the [release notes](https://github.com/tidyverse/tibble/tree/master/NEWS.md).
+
+## Tibble construction and validation
+
+End users should use the `tibble()` function to construct tibbles.
+It checks the input for consistency and makes sure that the returned tibble is valid.
+
+Package developers, however, can also consider the low-level `new_tibble()` constructor. Use `new_tibble()` to quickly construct a tibble from a list if you are very sure that the input is well-formed (i.e., a list of vectors of equal length).
+This function also supports the construction of subclasses of tibble through the `class` argument.
+
+In the development version of tibble, the `new_tibble()` constructor is a more faithful implementation of the [design advice for S3 classes given in Advanced R](https://adv-r.hadley.nz/s3.html#s3-classes).
+Specifically:
+
+- `new_tibble()` is very fast again and does very little checking itself.
+- The new `validate_tibble()` function is responsible for validating the structure of a tibble.
```

Comment:
If we plan on building the site prior to the release of this post, perhaps we should link to the function references?